File: examples/chat/producer-rust/src/producer/implementation/emitters/connected.rs
use super::{events, identification, producer::Control, Context, EmitterError};

pub async fn emit<E: std::error::Error>(
    identification: &mut identification::Identification,
    filter: &identification::Filter,
    context: &mut Context,
    control: &Control,
) -> Result<(), EmitterError> {
    events::connected::emit::<E>(identification, filter, context, control)
        .await
        .map_err(EmitterError::Emitting)
}

File: tests/workflow/producer/typescript (copy)/src/producer/events/connected.ts
import {
	Identification,
	Filter,
	Producer,
	Context,
	Protocol,
} from "../implementation/events";

export function emit(
	consumer: Identification,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<void> {
	context.connected(consumer.uuid());
	return Promise.resolve();
}
